app/controllers/images_controller.rb

```
@@ -37,19 +37,39 @@ def destroy
     image = Image.find_by(id: params[:id])
     if image
       image.destroy
-      flash[:success] = 'You have successfully deleted the image.'
+      flash[:success] = 'You have successfully deleted the images.'
     else
-      flash[:danger] = 'Unable to delete image.'
+      flash[:danger] = 'Unable to delete images.'
     end
     redirect_to images_path
   end
 
+  def share
+    @share_form = Images::ShareForm.new
+  end
+
+  def send_email
+    @share_form = Images::ShareForm.new(_share_form_params)
+    image_url = Image.find_by(id: params[:id]).image_link
+    if @share_form.valid?
+      ShareImageMailer.share_email(@share_form.recipient_email, image_url, @share_form.email_body).deliver_now
+      flash[:success] = 'You have successfully shared the images.'
+      redirect_to image_path(id: params[:id])
+    else
+      render :share
```

in this case, we could also return the failure status
i.e
`render :share, status: :unprocessable_entity`, and we can then assert this response in the test